upstairs/src/downstairs.rs

```diff
@@ -2019,6 +2031,8 @@ impl Downstairs {
 
         if self.clients[client_id].on_reconciliation_job_done(repair_id, next) {
             self.reconcile_current_work = None;
+            self.reconcile_repair_needed -= 1;
```

Comment:
Were we not handling `self.reconcile_repair_needed` correctly before?

Reply:
No, it got dropped in the great one task refactor and I failed to notice its absence.

Reply:
Got it, thanks for bringing it back!